gcp_variant_transforms/transforms/merge_header_definitions_test.py

```diff
@@ -0,0 +1,236 @@
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
+    headers_2 = self._get_vcf_header_from_reader(vcf_reader_2, 'file2')
+    combiner_fn = self._get_definitions_combiner_fn()
+
+    merged_definitions = combiner_fn.create_accumulator()
+    merged_definitions = combiner_fn.add_input(merged_definitions, headers_1)
+    merged_definitions = combiner_fn.add_input(merged_definitions, headers_2)
+    definitions = combiner_fn.extract_output(merged_definitions)
+    info_definitions = definitions.infos
+    format_definitions = definitions.formats
+
+    self.assertEqual(
+        format_definitions['NS'][Definition(1, 'Float')],
+        list(['file1']))
+    self.assertEqual(
+        format_definitions['DP'][Definition(2, 'Float')],
+        list(['file2']))
+    self.assertEqual(len(info_definitions), 0)
+
+  def test_merge_header_definitions_same_id_in_info_and_format_headers(self):
+    lines_1 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n']
+    lines_2 = [
+        '##fileformat=VCFv4.2\n',
+        '##FORMAT=<ID=NS,Number=1,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
+    headers_2 = self._get_vcf_header_from_reader(vcf_reader_2, 'file2')
+    combiner_fn = self._get_definitions_combiner_fn()
+
+    merged_definitions = combiner_fn.create_accumulator()
+    merged_definitions = combiner_fn.add_input(merged_definitions, headers_1)
+    merged_definitions = combiner_fn.add_input(merged_definitions, headers_2)
+    definitions = combiner_fn.extract_output(merged_definitions)
+    info_definitions = definitions.infos
+    format_definitions = definitions.formats
+
+    self.assertEqual(
+        info_definitions['NS'][Definition(1, 'Integer')],
+        list(['file1']))
+    self.assertEqual(
+        format_definitions['NS'][Definition(1, 'Float')],
+        list(['file2']))
+
+  def test_merge_header_definitions_save_five_copies(self):
+    lines_1 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=1,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n']
+    lines_2 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
```

Comment:
consider using a for loop and an array :)

Reply:
Done.